### ISSUE TITLE
netify errors

### ISSUE THREAD

**Issue by akomiqaia** [fac19/week8-DEKN#36]
`https://week8-dekn.netlify.app/dog?id=1` when we are trying to access the separate dog we can't get it on Netlify but it works fine on the local server. we had the same issue on Github pages.

**Comment by oliverjam** [comment]
I can tell you what's happening, but I don't know why. Your router isn't working on that link, which means you're making a full proper browser request for the URL. I don't know why the router isn't working though 😅 

This is a common problem with client-side routing. The browser requests `/dogs?id=1` from your server. Since this route only exists client-side the server won't find it, so you get a 404.

The way to fix it is to configure your server to _always_ serve `index.html` for any route it doesn't recognise. Then your client-side router can take over when the page JS loads and render the correct client-side route.

You can see how to configure Netlify to do this at the start of this article: https://www.netlify.com/blog/2020/04/07/creating-better-more-predictable-redirect-rules-for-spas/

**Comment by oliverjam** [comment]
Ah I figured it out!

Because you've got an `h2` _inside_ the `a` the `event.target` is `h2`, which means the router ignores this click and doesn't `preventDefault()`.

Generally you want inline elements like `a` or `strong` inside block level elements like headings. Putting the `a` inside the `h2` will fix this I think (although it's still worth configuring Netlify redirects like above, so refreshing the page doesn't cause 404s). 

This is an unfortunate limitation of how my basic router was implemented, sorry. Always use proper tested libraries in production! 😆
